pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -237,15 +237,14 @@ func (volumes *VolumesV3) expandVolume(volumeID string, newSize int) error {
 		NewSize: newSize,
 	}
 
-	// save initial microversion
-	mv := volumes.blockstorage.Microversion
+	// cinder online resize is available since 3.42 microversion
+	// https://docs.openstack.org/cinder/latest/contributor/api_microversion_history.html#id40
+	// this operation is thread-safe, since a dedicated client is initialized before the
+	// "expandVolume" method call
 	volumes.blockstorage.Microversion = "3.42"
```

Comment:
@kayrus As you already have PR to fix resize, I think we have one more issue here that needs to be fixed. hardcoding microversion here might not be correct , as for offline resize its not needed . we need to check before , if its online/offline resize. Do you have the setup to test the resize operation again ?

Reply:
@ramineni good catch, thank you. fixed.